plugins/ltac/g_auto.mlg

```diff
@@ -64,10 +64,10 @@ let eval_uconstrs ist cs =
 
 let pr_auto_using_raw _ _ _  = Pptactic.pr_auto_using Ppconstr.pr_constr_expr
 let pr_auto_using_glob _ _ _ = Pptactic.pr_auto_using (fun (c,_) ->
-    let _, env = Pfedit.get_current_context () in
+    let _, env = Pfedit.get_current_context () in (* XXX *)
     Printer.pr_glob_constr_env env c)
 let pr_auto_using _ _ _ = Pptactic.pr_auto_using
-    (let sigma, env = Pfedit.get_current_context () in
+    (let sigma, env = Pfedit.get_current_context () in (* XXX *)
```

Comment:
AFAICT this is the only `PRINTED BY` stuff that needs the evar map. @ppedrot is there something special about it? Does it mean the evarmap should be threaded everywhere, just for this one?

Reply:
Most likely. Virtually all runtime values depend on the current evar map (think about terms for instance).

Reply:
Anyways a lot of entries are missing a `PRINTED BY` declaration at all so I wonder how much these printers are used as for the missing printers we get a dummy value.

Reply:
`PRINTED BY` is almost never used in practice, except for a few base types that have a harcoded variant like terms. For it to be called, you'd need to use a tactic notation with the corresponding argument and try e.g. calling `idtac` on the bound variable. If you only use it for ML tactics, you're most likely going to pass it directly to some ML code, which means that you won't ever put it inside a Ltac environment that would be used for debug printing.

Reply:
And also, thinking about it, for this to be actually used, you need not to have the `TYPED AS` clause, otherwise the runtime representation of that generic argument will be the same at the one given in that clause, resulting in a call to that printer instead. I am not even sure we have an instance of that in the Coq code.

(Note that `PRINTED BY` would be used nonetheless if there is no `GLOB/RAW PRINTED BY` clause, in which case the argument printer is used for all three levels.)

Reply:
So, what should I do here, in your opinion @ppedrot ?